johnpapa/vscode-peacock
# Live Share Integration

# Integration with Live Share

![ezgif com-video-to-gif](https://user-images.githubusercontent.com/1478800/57961665-5f0ba580-78c5-11e9-9da5-fb0c5555bc7a.gif)

## Purpose

The PR adds integration with [Live Share](https://marketplace.visualstudio.com/items?itemName=MS-vsliveshare.vsliveshare) extension. When a user has [Live Share](https://marketplace.visualstudio.com/items?itemName=MS-vsliveshare.vsliveshare) alongside with the [Peacock](https://marketplace.visualstudio.com/items?itemName=johnpapa.vscode-peacock), [Peacock](https://marketplace.visualstudio.com/items?itemName=johnpapa.vscode-peacock) will detect that and enable two additional command palette commands: `Change Color for Live Share Host` and `Change Color for Live Share Guest`. The commands allow users to pick a color from the favorites that can be used during [Live Share](https://marketplace.visualstudio.com/items?itemName=MS-vsliveshare.vsliveshare) session.

## What

[Live Share](https://marketplace.visualstudio.com/items?itemName=MS-vsliveshare.vsliveshare) is a VSCode extension that adds collaboration capabilities to the editor. Using [Live Share](https://marketplace.visualstudio.com/items?itemName=MS-vsliveshare.vsliveshare) one can share their local files, terminals, local servers and debugging sessions with someone remote.

## How to Test

1. Install [Live Share](https://marketplace.visualstudio.com/items?itemName=MS-vsliveshare.vsliveshare) and reload the VSCode instance.
2. Open a folder in the VSCode instance.
3. Open `command palette` and type `Change Color for Live Share Host`. Select the color of interest from the list.
4. Open `command palette` and type `>Live Share: StartCollaboration Session (Share)` to share the current workspace.
5. Notice that as soon as the session starts the color of the workspace will change to the previously selected one.

## What to Check

[included in unit tests]

1. After Live Share session is finished, VSCode instance should have the same color as before the session started (or absence of color).
2. After Live Share session is finished, VSCode instance should have the same color as before the session started (or absence of color), even if a user changed the color of Live Share session multiple times.
3. If a Live Share session color is changed during Live Share session it should be automatically reflected in the UI.

## Checklist of changes included

- [x] Prettier formatting - The project does not have
- [x] console.log removed
- [x] Dead code removed
- [x] Unnecessary comments removed
- [x] Images updated (if applicable)
